pkg/registry/core/pod/etcd/eviction.go

```diff
@@ -116,6 +116,10 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 }
 
 func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb policy.PodDisruptionBudget) (ok bool, err error) {
+	if pdb.Status.ObservedGeneration == nil || *pdb.Status.ObservedGeneration != pdb.Generation {
```

Comment:
I talked with @wojtek-t and we initially agreed that there is optimistic locking implemented in client/etcd that would make the extra generation not needed. However now he has some second thoughts how it exactly works and if there are retries on version mismatch. We are checking it right now.